### DISCUSSION TITLE
How do I generate a JWT auth token as an admin as a golang or JS framework app?

### DISCUSSION THREAD

**Discussion by xrd** [pocketbase/pocketbase#6029] [Q&A] [answered]
I want to extend pocketbase using the JS framework interface (but I can probably do it with golang if that's easier for some reason). 

I would like to add a route to the pocketbase server `/internal/auth` (which I will protect via a firewall) that allows a user to provide a secret and an email and then retrieve a JWT auth token for that user. I can then pass this auth token back to the client side JS and have it operate as if the user logged in normally. I don't want to keep track of the actual password for each user; I want to assume that if someone reaches this `/internal/auth` path then they are privileged and can generate a JWT token for whatever email they provide.

I don't see a way to do this when I review the docs here: https://pocketbase.io/docs/js-overview/#javascript-engine

Is there a way to do this using the existing API?

**Comment by xrd** [comment, marked as answer]
Hmm, maybe this is the way to do it?

https://pocketbase.io/docs/authentication/#users-impersonation

**Comment by ganigeorgiev** [comment]
I'm not entirely sure that I understand the description and the desired behavior.

If you want to protect a custom route, you can check the available builtin middlewares in https://pocketbase.io/docs/js-routing/#builtin-middlewares.

If you want to generate record tokens programmatically, then you can check https://pocketbase.io/docs/js-records/#generating-and-validating-tokens (_the same link also contains examples how to retrieve a record_).

Note that the above links works only for PocketBase v0.23+. For the older versions you can check https://pocketbase.io/old/docs (for how to upgrade, please refer to the [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#v0230)).
